src/Extensions/Services.Remoting/Internal/Client/ServiceRemotingClientWrapper.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Fabric;
+using System.Threading.Tasks;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+using Microsoft.ServiceFabric.Services.Remoting.V2;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
+
+namespace Microsoft.Omex.Extensions.Services.Remoting.Client
+{
+	/// <summary>
+	/// Wrapper for adding Omex headers to Remoting request
+	/// </summary>
+	internal class ServiceRemotingClientWrapper : IServiceRemotingClient
+	{
+		/// <summary>
+		/// Suffix for this class activities
+		/// </summary>
+		/// <remarks>
+		/// Should end with RequestOut to be enabled by out telemetry
+		/// </remarks>
+		private const string RequestOutActivitySuffix = "RequestOut";
+
+		private const string OneWayMessageActivityName = Diagnostics.ListenerName + "OneWay" + RequestOutActivitySuffix;
+
+		private const string RequestActivityName = Diagnostics.ListenerName + RequestOutActivitySuffix;
+
+		private readonly DiagnosticListener m_diagnosticListener;
+
+		internal readonly IServiceRemotingClient Client;
+
+		public ServiceRemotingClientWrapper(IServiceRemotingClient client, DiagnosticListener? diagnosticListener = null)
+		{
+			Client = client;
+			m_diagnosticListener = diagnosticListener ?? Diagnostics.DefaultListener;
```

Comment:
>Diagnostics.DefaultListener; [](start = 48, length = 28)

General comment for defaults - having lots of defaults makes it difficult to troubleshoot a problem as it is hard to see in logs which value was used. I think avoiding defaults as much as possible makes it easier for debugging services.

Reply:
I added the ability to overload listener just for unit test purpose and it's actually used to log events, but it might be a good idea of adding some clarifications here, I'll try to add them in future pull requests